### PR TITLE
Auton arm

### PR DESCRIPTION
**Pull request by rnagpal02**

### CODE REVIEW

**Review by mmartin4972 — APPROVED** [review]

File: jetson/nav/auton_arm/stateMachine.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    //TODO implement if we want 
+} // publishNavState()
+
+AutonArmState AutonArmStateMachine::executeOff() { 
+    if( mPhoebe->roverStatus().autonState().is_auton )
+    {
+        cout << "Reached execute off" << endl;
+        //TODO add what we want here
+    }
+    return AutonArmState::Off;
```
Comment (aswarner) [inline jetson/nav/auton_arm/stateMachine.cpp:136]: this function will need a stop() function to stop all controls to the arm, maybe mPhoebe->arm->stop(), to guarantee that teleop does not continue receiving commands to move. 

alternatively teleop might check if this auton program is publishing on/off, but our publish function is empty

File: jetson/nav/auton_arm/stateMachine.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+} // AutonArmStateMachine Destructor
+
+void AutonArmStateMachine::run() {
+    publishNavState();
+    if( isRoverReady() )
+    {
+        mStateChanged = false;
+        AutonArmState nextState = AutonArmState::Unknown;
+
+        if( !mPhoebe->roverStatus().autonState().is_auton )
```
Comment (aswarner) [inline jetson/nav/auton_arm/stateMachine.cpp:40]: we will eventually change mPhoebe to mRover and rover

File: jetson/nav/auton_arm/stateMachine.cpp [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+    mLcmObject.publish("/target_orientation", &orientation);
+    cout << "Sent target position\n";
+    
+    return AutonArmState::Done;
+} // executeSendCoordinates()
+
+bool AutonArmStateMachine::isEqual(const TargetPosition &a, const TargetPosition &b) const {
+    return sqrt( pow(a.x-b.x, 2) + pow(a.y-b.y, 2) + pow(a.z-b.z, 2) ) < MARGIN_OF_ERROR;
+} // isEqual( TargetPosition )
```
Comment (aswarner) [inline jetson/nav/auton_arm/stateMachine.cpp:219]: all c++ files will need newlines

File: jetson/nav/auton_arm/stateMachine.cpp [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+bool AutonArmStateMachine::isRoverReady() const {
+    return mStateChanged || // internal data has changed
+           mPhoebe->updateRover( mNewRoverStatus ) || // external data has changed
+           isTagReceived || // has a new tag been received 
+           mPhoebe->roverStatus().currentState() == AutonArmState::PauseTag; // if timer is active
+} //isRoverReady()
+
+void AutonArmStateMachine::publishNavState() const
+{
+    //TODO implement if we want
```
Comment (aswarner) [inline jetson/nav/auton_arm/stateMachine.cpp:127]: implement

File: jetson/nav/auton_arm/stateMachine.cpp [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+AutonArmState AutonArmStateMachine::executeDone() {
+    cout << "Done" << endl;   
+    return AutonArmState::Done;
+} // executeDone()
+
+AutonArmState AutonArmStateMachine::executeWaitingForTag() {
+    // If statement to check if tag recieved
+    if(!isTagReceived) {
+        cout << "Waiting for tag" << endl;
```
Comment (aswarner) [inline jetson/nav/auton_arm/stateMachine.cpp:147]: this will print a LOT. good for now, will change later

**Pull request merged**